source/_posts/Docker-Swarm.md

```diff
@@ -0,0 +1,346 @@
+
+Om te berekenen hoeveel managers er mogen uitvallen voordat het cluster niet meer kan functioneren wordt de volgende berekening gebruikt: (N-1/2).
+Bij dus een cluster van 3 manager mag er 1 manager uitvallen en zal het cluster nog steeds functioneren. Docker adviseert om niet meer dan 7 managers te gebruiken om performance issues met het synchroniseren te voorkomen. Voor meer informatie zie https://docs.docker.com/engine/swarm/raft/
+
+Om een Docker Swarm cluster op te zetten kun je de volgende stappen uitvoeren:
+- Installeer 5 virtuele machine met bijv. Ubuntu server. Voor de 3 managers hebben we niet hele zware virtuele machines nodig. Een basic A1 1.75 GB RAM volstaat al voor een manager node. 
+Voor de 2 worker nodes zou ik kiezen voor een virtuele machine met 8 GB RAM. 
+- Installeer Docker community edition op alle 5 de servers. Volg de handleiding op https://docs.docker.com/install/linux/docker-ce/ubuntu/. 
+- Op 1 van de manager servers voer het volgende commando uit om Docker Swarm te initialiseren.
+```
+$ docker swarm init --advertise-addr "public ipadres van de server"
+Swarm initialized: current node (bvz81updecsj6wjz393c09vti) is now a manager.
+
+To add a worker to this swarm, run the following command:
+
+    docker swarm join \
+    --token SWMTKN-1-3pu6hszjas19xyp7ghgosyx9k8atbfcr8p2is99znpy26u2lkl-1awxwuwd3z9j1z3puu7rcgdbx \
+    172.17.0.2:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+```
+- Voer het bovenstaande docker swarm join token uit op de 2 nodes. 
+- Voer het comamndo docker swarm join-token manager uit en voer het join commando uit op de overige 2 managers.
+
+Als dit klaar gedaan is heb je een Docker Swarm cluster gemaakt zoals in het onderstaande overzicht is weergegeven.
+
+<img src="/images/Docker-Swarm-overview.png" />
+Mocht je meer details willen hebben over het opzetten van een Docker Swarm cluster kijk dan op: https://docs.docker.com/engine/swarm/swarm-tutorial/create-swarm/
+
+Docker Swarm zal niet automatisch schalen als de load op je applicatie hoger wordt, wil je meer containers uitrollen van je applicatie kun je dit doen door meer replica's aan te maken. Als je bijv. de service api_api wilt opschalen naar 5 instanties kun je dit doen door middel van het volgende commando:
+
+```
+$ docker service scale api_api=5 
+```
+Je kunt de schaling ook regelen in de UI van Portainer.
```

Comment:
Zeg even erbij dat later ingaat op wat portainer precies is

Reply:
Je kunt de schaling ook regelen in de UI van Portainer. Verderop in het artikel ga ik dieper in op hoe en wat Portainer is.